### PR TITLE
Observation registry cleanup

### PR DESCRIPTION
**Pull request by msangoi**
Observation registry API cleanup and renaming.

### CODE REVIEW

File: leshan-server-core/src/main/java/org/eclipse/leshan/server/LwM2mServer.java [hunk 1]
```diff
@@ -115,7 +115,7 @@
      * Get the Observation registry containing of current observation. You can use this object for listening resource
```
Comment (sbernard31) [inline leshan-server-core/src/main/java/org/eclipse/leshan/server/LwM2mServer.java]: _"registry"_ detected !

File: leshan-server-core/src/main/java/org/eclipse/leshan/server/observation/ObservationService.java [hunk 2]
```diff
@@ -21,17 +21,9 @@
 import org.eclipse.leshan.server.client.Registration;
 
 /**
- * A registry for keeping track of observed resources implemented by LWM2M Clients.
- * 
+ * A service for keeping track of observed resources implemented by LWM2M Clients.
```
Comment (jvermillard) [inline leshan-server-core/src/main/java/org/eclipse/leshan/server/observation/ObservationService.java]: This javadoc doesn't mention the "cancel".

Proposal:
A service keeping track observation. Can be used for finding observations and cancel them.

File: leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/RegistrationTest.java [hunk 3]
```diff
@@ -213,7 +213,7 @@ public void register_observe_deregister_observe() throws NonUniqueSecurityInfoEx
 
         // check observation registry is not null
```
Comment (sbernard31) [inline leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/RegistrationTest.java]: _"registry"_ detected !

File: leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/LwM2mResponseBuilder.java [hunk 4]
```diff
@@ -277,7 +277,7 @@ public void visit(final ObserveRequest request) {
                 // observe request succeed so we can add and observation to registry
```
Comment (sbernard31) [inline leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/LwM2mResponseBuilder.java]: _"registry"_ detected !

**Comment by sbernard31** [comment]
More _"registry"_ detected ;) 

About Observations:
  - 3 in `CaliforniumObservationTest` too.
  - 1 in `LwM2mServer`.
 
About Registrations (so not linked to this PR): 
  - 1 in `RedisRegistrationEventPublisher`.
  - 1 in `QueuedRequestSender`.
  - several in `CoAPMessageTracer`.

**Pull request merged**